search-commons/src/main/java/no/unit/nva/search/constants/ApplicationConstants.java

```diff
@@ -63,4 +70,37 @@ private static String readSearchInfrastructureApiUri() {
     private static String readSearchInfrastructureAuthUri() {
         return ENVIRONMENT.readEnv("SEARCH_INFRASTRUCTURE_AUTH_URI");
     }
+
+
+    private static TermsAggregationBuilder generateSimpleAggregation(String term, String field) {
+        return AggregationBuilders
+                .terms(term)
+                .field(field)
+                .size(100);
+    }
+
+    private static NestedAggregationBuilder generateObjectLabelsAggregation(String object) {
+        return new NestedAggregationBuilder(object, object)
+                .subAggregation(generateIdAggregation(object));
+    }
+
+    private static TermsAggregationBuilder generateIdAggregation(String object) {
+        return new TermsAggregationBuilder(ID)
+                .field(jsonPath(object, ID))
+                .size(100)
+                .subAggregation(generateLabelsAggregation(object));
+    }
+
+    private static NestedAggregationBuilder generateLabelsAggregation(String object) {
+        return new NestedAggregationBuilder(LABELS, jsonPath(object, LABELS))
+                .subAggregation(generateSimpleAggregation(ENGLISH,  jsonPath(object, LABELS, ENGLISH, KEYWORD)))
+                .subAggregation(generateSimpleAggregation(BOKMAAL, jsonPath(object, LABELS, BOKMAAL, KEYWORD)))
+                .subAggregation(generateSimpleAggregation(NYNORSK, jsonPath(object, LABELS, NYNORSK, KEYWORD)))
+                .subAggregation(generateSimpleAggregation(SAMI, jsonPath(object, LABELS, SAMI, KEYWORD)));
```

Comment:
You call jsonPath 4 times with 3 of the arguments beeing the same, and the only one that varies is also called to generateSimpleAggregation directly. Any simpler way of writing this?

perhaps a generateSimpleAggregationForLanguage(String language), that calls generateSimpleAggregation with the needed jsonPath?